### PR TITLE
REDDEV-510 Refactor to add form for adding counts

### PR DESCRIPTION
**Pull request by wsams**
Refactored to support a form for adding report counts.

settings.php no longer persists sample configuration if not present.

Extracted ReportSummaryForm component.

Default view is an empty page with "Create a Report" button. Clicking
the button opens a form for entering a title and a report. Upon save the
report configuration is saved by settings.php. Either saves new
configuration to the database, or appends to the existing config.

Currently the form remains on the screen and multiple summaries can be
added.

Removed sample report config and updated documentation and tests.

### CODE REVIEW

**Comment by wsams** [comment]
@kirenotneb , @lawhead , this PR contains all of the refactor to add a form to create report summaries (a node of the report with a total count)

Since it's a bit large I also broke this work up into separate branches and created 4 pull requests. If you take a look at the pull requests you'll see four additional PRs. The title is prefixed with a number to represent the order. If you prefer to review in steps we can close this one and review the 4 PRs separately. Preference?

**Comment by wsams** [comment]
Here are some screenshots.

screen shot 2018-09-18 at 16 29 47

screen shot 2018-09-18 at 16 29 54

screen shot 2018-09-18 at 16 30 37

File: lib/settings.php [hunk 1]
```diff
@@ -11,8 +11,29 @@
...
+    if ($_GET['action'] === 'saveReportSummary') {
+        $requestBody = trim(file_get_contents('php://input'));
 
-print($settings);
+        if (!empty($requestBody)) {
+            $data = json_decode($requestBody, true);
+            $reportSummary = json_decode($data['reportSummary'], true);
+
+            $reportConfig = new ReportConfig($project_id, $module);
+            $reportConfig->saveReportSummary($reportSummary['reportSummary']);
```
Comment (kirenotneb) [inline lib/settings.php]: You may want to add some validation of structure of the summary to ensure it is actually what we believe it is.

**Comment by kirenotneb** [comment]
I can sympathize that we want to keep PRs very specific. But when you are adding a significant piece of functionality it may be hard to break it up into small focused PRs. I think my preference would be to have a single branch with multiple commits of each aspect. So in this case you would have 1 PR with 4 commits associated with it.

**Pull request merged**